tapdb/assets_store.go

```diff
@@ -2728,6 +2739,70 @@ func (a *AssetStore) QueryProofTransferLog(ctx context.Context,
 	return timestamps, err
 }
 
+// ConfirmProofDelivery marks a transfer output proof as successfully
+// delivered to counterparty.
+func (a *AssetStore) ConfirmProofDelivery(ctx context.Context,
+	args tapfreighter.ProofDeliveryConfirmation) error {
```

Comment:
Actually, this should become way easier with the other comment I made. Main point here is to use the output's `Anchor.Outpoint` instead of just the `anchorTXID` and the new `position` to make this update bullet proof.